server/build/plugins/watch-pages-plugin.js

```diff
@@ -53,7 +53,7 @@ export default class WatchPagesPlugin {
 
         if (compiler.hasEntry(name)) return
 
-        const entries = [hotMiddlewareClientPath, f]
+        const entries = [hotMiddlewareClientPath, f + '?entry']
```

Comment:
Since we check if it's an entry page using the value on `../loaders/hot-self-accept-loader.js` https://github.com/zeit/next.js/blob/master/server/build/loaders/hot-self-accept-loader.js#L15

We do this check for https://github.com/zeit/next.js/issues/294